.github/workflows/autobuild.yml

```diff
@@ -65,100 +88,140 @@ jobs:
   release_assets:
     name:                           Build assets for ${{ matrix.config.config_name }}
     needs:                          create_release
-    runs-on:                        ${{ matrix.config.building_on_os }}
     strategy:
       fail-fast:                    false
       matrix: # Think of this like a foreach loop. Basically runs the steps with every combination of the contents of this. More info: https://docs.github.com/en/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
         config:
-           - config_name:           Linux (normal)
-             target_os:             linux
-             building_on_os:        ubuntu-18.04
-             asset_path:            deploy/Jamulus_amd64.deb
-             asset_name:            jamulus_${{ needs.create_release.outputs.version_name }}_ubuntu_amd64.deb
-           - config_name:           Linux (headless)
+           - config_name:           AndroidAPK (artifact+codeQL)
+             target_os:             android
+             building_on_os:        ubuntu-20.04
+             cmd1_prebuild:         "sh ./autobuild/android/autobuild_apk_1_prepare.sh "
+             cmd2_build:            "./autobuild/android/autobuild_apk_2_build.sh "
+             cmd3_postbuild:        "./autobuild/android/autobuild_apk_3_copy_files.sh "
+             uses_codeql:           true
+
+           - config_name:           Linux (artifacts+codeQL)
              target_os:             linux
-             building_on_os:        ubuntu-18.04
-             asset_path:            deploy/Jamulus_headless_amd64.deb
-             asset_name:            jamulus_headless_${{ needs.create_release.outputs.version_name }}_ubuntu_amd64.deb
-           - config_name:           MacOS
+             building_on_os:        ubuntu-20.04
+             cmd1_prebuild:         "sh ./autobuild/linux/autobuild_deb_1_prepare.sh"
+             cmd2_build:            "./autobuild/linux/autobuild_deb_2_build.sh"
```

Comment:
Why no "sh" here?

Reply:
Fixed.

Reply:
the scripts 2&3 need "bash" for some of the commands.
invoking 1 with bash leads to some permission-errors

using sh for step one and bash for the other two works well.

Reply:
added a comment

Reply:
Ok. Great.